std/assembly/string.ts

```diff
@@ -484,6 +486,118 @@ import { idof } from "./builtins";
     // releases result
   }
 
+  toLowerCase(): String {
+    var len = <usize>this.length;
+    if (!len) return this;
+    var codes = __alloc(len * 2 * 2, idof<String>());
+    var j: usize = 0;
+    for (let i: usize = 0; i < len; ++i, ++j) {
```

Comment:
Looks like removing the incrementor `++i, ++j` here and doing this in the respective cases below can save us one operation.

Reply:
Yes, but this significantly increase code size.

Reply:
Hmm, would also help in spotting potential OOB accesses, which is what I was initially looking for. Wdyt, would making it a bit faster and clearer at the same time justify changing it?

Reply:
In first variant I used exactly suggested by you version, but I rewrote to current due to better code size. In other case all other branches need add increments for `i` / `j`

Reply:
Wondering how much of a difference this is. Especially considering that we are including relatively large tables anyway if these functions are used, this appears to be negligible to me. Also appears that an optimizer, at some point, might optimize to what we'd write manually today.

Reply:
https://github.com/MaxGraey/assemblyscript/compare/case-mapping...MaxGraey:refactored-casemap?expand=1

Reply:
All right I could you that version. Mo big deal, but I'm not sure this got some performance benefits

Reply:
Current version: https://webassembly.studio/?f=ul2gpz0nln
```
AS toLowerCase: 36.135009765625ms
AS toUpperCase: 34.879150390625ms
JS toLowerCase: 69.303955078125ms
JS toUpperCase: 65.5810546875ms
```

Suggested version: https://webassembly.studio/?f=mzyue5tfr9o
```
AS toLowerCase: 35.564208984375ms
AS toUpperCase: 36.23779296875ms
JS toLowerCase: 79.428955078125ms
JS toUpperCase: 84.647216796875ms
```

Reply:
Btw on Firefox 71 suggested version 2.5x slower than current. So I prefer leave everithing as is